idl/chimera.pro

```diff
@@ -0,0 +1,535 @@
+printf,2,formtab[1]
+
+;=====contours the identified datapoints=======
+Contour,def,ax,ay,/over,levels=[0.5,1.5],path_xy=xy,path_info=info,/path_data_coords
+
+;=====cycles through contours=========
+for i=0L,(n_elements(info)-1) do begin
+
+;=====only takes values of minimum surface length and calculates area======
+	if info(i).n gt 100 then begin
+
+		area=poly_area(xy[0,(info(i).offset):(info(i).offset)+(info(i).n)-1],xy[1,(info(i).offset):(info(i).offset)+(info(i).n)-1])
+		arcar=(area*dattoarc)*dattoarc
+		if arcar gt 1000 then begin
+
+;=====finds centroid=======
+		chpts = [FINDGEN(info(I).N), 0]
+		cent=[mean(xy[0,info(i).offset+chpts]),mean(xy[1,info(i).offset+chpts])]			
+
+;===remove quiet sun regions encompassed by coronal holes======
+			if def[max(xy[0,info(i).offset+chpts])+1.,xy[1,info(i).offset+min(where((xy[0,info(i).offset+chpts]) eq max(xy[0,info(i).offset+chpts])))]] gt 0 and iarr[max(xy[0,info(i).offset+chpts])+1.,xy[1,info(i).offset+min(where((xy[0,info(i).offset+chpts]) eq max(xy[0,info(i).offset+chpts])))]] gt 0 then begin
+				subscripts=POLYFILLV(xy(0,INFO(I).OFFSET + chpts ),xy(1,INFO(I).OFFSET + chpts ),s[1],s[2])
+				iarr[subscripts]=0
+			endif else begin
+
+;====create a simple centre point======
+				arccent0=coord[0,(cent[0]),0]
+				arccent1=coord[1,0,(cent[1])]
+				
+;=====finds coordinates of CH boundaries=======
+				Xwb=coord[0,(max(xy[0,info(i).offset+chpts])),0]
+				Ywb=coord[1,(max(xy[0,info(i).offset+chpts])),xy[1,info(i).offset+min(where((xy[0,info(i).offset+chpts]) eq max(xy[0,info(i).offset+chpts])))]]
+				Xeb=coord[0,(min(xy[0,info(i).offset+chpts])),0]
+				Yeb=coord[1,(min(xy[0,info(i).offset+chpts])),xy[1,info(i).offset+min(where((xy[0,info(i).offset+chpts]) eq min(xy[0,info(i).offset+chpts])))]]
+				Ynb=coord[1,0,(max(xy[1,info(i).offset+chpts]))]
+				Xnb=coord[0,xy[0,info(i).offset+min(where((xy[1,info(i).offset+chpts]) eq max(xy[1,info(i).offset+chpts])))],(max(xy[1,info(i).offset+chpts]))]
+				Ysb=coord[1,0,(min(xy[1,info(i).offset+chpts]))]
+				Xsb=coord[0,xy[0,info(i).offset+min(where((xy[1,info(i).offset+chpts]) eq min(xy[1,info(i).offset+chpts])))],(min(xy[1,info(i).offset+chpts]))]
+
+				eastl=lon[where(coord[0,*,0] eq xeb),where(coord[1,0,*] eq yeb)]
+				westl=lon[where(coord[0,*,0] eq xwb),where(coord[1,0,*] eq ywb)]
+				width=westl-eastl
+
+;====classifies off limb CH regions========
+				if (arccent0^(2)+arccent1^(2)) gt rs^(2) or (coord[0,xy(0,INFO(I).OFFSET),0]^(2) + (coord[1,0,xy(1,INFO(I).OFFSET)])^(2)) gt rs^(2) then begin
+					subscripts=POLYFILLV(xy(0,INFO(I).OFFSET + chpts ),xy(1,INFO(I).OFFSET + chpts ),s[1],s[2])
+					offarr[subscripts]=1
+				endif else begin
+
+;=====classifies on disk coronal holes=======
+					subscripts=POLYFILLV(xy(0,INFO(I).OFFSET + chpts ),xy(1,INFO(I).OFFSET + chpts ),s[1],s[2])
+					iarr[subscripts]=ident
+
+;====create an array for magnetic polarity
+					poslin=where(iarr eq ident)
+					pos=fltarr(n_elements(poslin),2)
```

Comment:
This where is guaranteed to not be -1 as it is the number of pixels occupied for a feature. For the code to get to this point it is required that its size be gt 4000 pixels squared. I can add the verification that it is not -1 but that seems redundant